Models/ProductsListPart.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Utilities;
+using Orchard.Environment.Extensions;
+using Orchard.Security;
+using Newtonsoft.Json.Linq;
+
+namespace Nwazet.Commerce.Models {
+    [OrchardFeature("Nwazet.PersistentCart")]
+    public class ProductsListPart : ContentPart {
+        //This part goes in the infoset
+
+        private List<ShoppingCartItem> _items {
+            get {
+                var fromInfoset = Retrieve<string>("SerializedItems");
+                if (string.IsNullOrWhiteSpace(fromInfoset)) {
+                    return null;
+                }
+                var jsonInfoset = JObject.Parse("{'List':" + fromInfoset + "}");
+                return jsonInfoset["List"].Children()
+                    .Select(jt => {
+                        var item = new ShoppingCartItem(
+                            productId: int.Parse(jt["ProductId"].ToString()),
```

Comment:
`jt["ProductId"]` here is a `Newtonsoft.Json.Linq.JToken`. It does not cast directly to int. Its `ToString()` gives em the string that in the json is the value for the parameter. THen I can parse that to int.